ImagePull.py

```diff
@@ -1,192 +1,124 @@
-from tkinter.simpledialog import askstring
-from operator import eq, ge, le, lt, gt
+import argparse
+import datetime
 from PIL import Image
 from PIL.ExifTags import TAGS
 from csvwriter import write_report
 
-# returns an int that contains only the numbers that are found in the file name that is passed in or None
-def configure(cat_num):
-    if cat_num:
-        try:
-            return int("".join([n for n in cat_num if n.isdigit()]))
-        except ValueError:
-            return None
-    return None
-
-# coordinates search function passing in appropriate info and operator expressions according to the drop selections
-def create_image_list(r1, r2, md,source,md_dict,op_dict,md_drop,op_drop):
-    if r2:
-        if r1:
-            if md:
-                return md_search(range_file_search(r1, r2,source), md,md_dict,md_drop)
-            return range_file_search(r1, r2,source)
-        if md:
-            return md_search(single_file_search(r2, op_dict[op_drop],source), md,md_dict,md_drop)
-        return single_file_search(r2, op_dict[op_drop],source)
-    if md:
-        return md_search(source_search(source), md,md_dict,md_drop)
-    return None
+def copy_files(image_paths, destination):
+    # Creates a new directory and copies images into it, returns path of new directory or False
+    folder = "ImagePull[%s]" % (datetime.date.today())
+    new_dir = str(destination % folder)
+    try:
+        os.mkdir(new_dir)
+    except FileExistsError:
+        return False
+    for path in image_paths:
+        shutil.copy(path, new_dir)
+    return new_dir
 
-# builds dict of metadata from files passed in, searches dict and returns list of files that contain the specified MD
-def md_search(files, md,md_dict,md_drop):
-    file_list = []
-    for file_name in files:
-        exif = Image.open(file_name, "r")._getexif()
-        if exif:
-            meta_data = [v for (k, v) in exif.items() if TAGS.get(k) == md_dict[md_drop]]
-            if meta_data:
-                if md.lower() in meta_data[0].decode("utf-16").lower():
-                    file_list += [file_name]
-    return file_list
-
-# returns a list of paths for each .JPG file in the source directory
 def source_search(source):
-    return [os.path.join(path, f) for path, sub, files in os.walk(source) for f in files if (f.endswith(".JPG"))]
-
-# searches source directory list for file names contained by the operator passed in for the single file name entered
-def single_file_search(image, op,source):
-    return [file for file in source_search(source) if op(configure(os.path.basename(file)), int(image))]
-
-# searches source list for file names in range of file names entered
-def range_file_search(image1, image2,source):
-    return [file for file in source_search(source) if
-            ge(configure(os.path.basename(file)), int(image1)) and le(configure(os.path.basename(file)), int(image2))]
+    # Returns a list of each .JPG file path in the source directory
```

Comment:
do we know for sure that all of the image files have extension uppercase .JPG (as opposed to .jpg or .jpeg or other extensions?)

Reply:
They will be of "jpeg" type, I have updated to flag files with endings as `.JPG/.jpg/.JPEG/.jpeg`